### DISCUSSION TITLE
rfsrc: Family assigned surv-CR when it is supposed to be surv

### DISCUSSION THREAD

**Discussion by vfincke** [kogalur/randomForestSRC#415] [General]
Hey there, 

I am trying to build a random forest for survival with right censored data. In my data table I have numeric values for 100 variables, the status (V1, coded as 0=alive, 1=dead) and the OS in years (V2). When doing: 
> rfModel <- rfsrc(Surv(V2, V1) ~.,
+                         data = dataS,
+                         importance = TRUE)

the output gives Family: surv-CR, although this is supposed to be a normal survival forest. 
Can you tell me, what is going on here? I did not find an option to set the family manually. 

Thanks in advance and best wishes, 

Victoria 

Here the full output: 
> print(rfModel)
                         Sample size: 81
                    Number of events: 36, 45
                     Number of trees: 500
           Forest terminal node size: 15
       Average no. of terminal nodes: 3.486
No. of variables tried at each split: 10
              Total no. of variables: 100
       Resampling used to grow trees: swor
    Resample size used to grow trees: 51
                            Analysis: RSF
                              Family: surv-CR
                      Splitting rule: logrankCR *random*
       Number of random split points: 10
   (OOB) Requested performance error: 0.47142857, 0.41818182

**Comment by ishwaran** [comment]
I believe your censoring variable "V1" is a factor with levels 0 and 1.  The code is converting this to a numeric value which are values 1 and 2 which then triggers a competing risk analysis where there are 2 competing risks (and no censored data as you can see by the fact that number of events sum to the total sample size 81).

I am moving to discussion since this is not an issue.

**Reply by vfincke** [reply]
Thank you very much for the quick reply! 
V1 is in fact in my data a factor. How can I dress this issue then? Shall I code the status as something different?

**Reply by vfincke** [reply]
I figured it out - thanks again!!

**Reply by bessierue** [reply]
vfincke, please tell me how you adressed this as i am also faced with a similar problem

**Reply by vfincke** [reply]
Sure, I converted the status column from factor to numeric before doing the random forest with it.
Data$status <- as.numeric(levels(Data$status))[Data$status]

**Reply by bessierue** [reply]
Thank you for the response. This is the structure of my status variable. SURVSTAT            : Factor w/ 2 levels "CENSORED","LTFU": 2 1 1 1 1 1 2 2 2 1 .  Do i first set the coding to 0,1 from 2,1 then convert to numeric or i just convert to numeric without first setting to 0,1

**Reply by vfincke** [reply]
It has to be 0 and 1 for rfscr() to recognize it as surv. If there are 2s it will change to survival-CR

**Reply by bessierue** [reply]
thank you. let me try it

**Comment by bessierue** [comment]
vfincke, please tell me how you adressed this as i am also faced with a similar problem

**Comment by ishwaran** [comment]
`as.numeric(V1)-1`

If it's a factor with two levels, `as.numeric` will convert to 1 and 2 so you need to subtract by 1

```
> V1=factor(c(1,1,1,0,1,0))
> V1
[1] 1 1 1 0 1 0
Levels: 0 1

> as.numeric(V1)
[1] 2 2 2 1 2 1
> as.numeric(V1)-1
[1] 1 1 1 0 1 0
```

**Reply by bessierue** [reply]
thank you. Please assist with this code. I am trying to print prediction error curves for 4 models on same plot and i dont want the reference curve to show. what should i do? 
brier <- pec(list("LogRankScore"=f1,"LogRank"=f2,"bs.Gradient"=f3,"CoxPH"=f4),data=TEST_data,formula=Surv(SURVTIMEAC, SURVSTAT)~1,include.time = TRUE  )
print(brier)
plot(brier, type = 'l', ylim = c(0, 1)). Also how do i adjust the position on the legend in the same code. I dont seem to be getting joy .

**Comment by bessierue** [comment]
Hallo.
Please assist with this code. I am trying to print prediction error curves
for 4 models on same plot and i dont want the reference curve to show. what
should i do?
brier <-
pec(list("LogRankScore"=f1,"LogRank"=f2,"bs.Gradient"=f3,"CoxPH"=f4),data=TEST_data,formula=Surv(SURVTIMEAC,
SURVSTAT)~1,include.time = TRUE  )
print(brier)

Regards,

Bester

plot(brier, type = 'l', ylim = c(0, 1))
